### ISSUE TITLE
Line style picker appearing offset from where the drop down should be

### ISSUE THREAD

**Issue by K4LCIFER** [xournalpp/xournalpp#3794]
(Please complete the following information, and then delete this line)

**Affects versions :**
 - OS: Arch Linux 
 - (Linux only) Desktop environment: Gnome Wayland
 - Which version of libgtk do you use: 3.24.31
 - Version of Xournal++: 1.1.0
 - Installation method: Official Arch Linux repository release

**Describe the bug**
Clicking the drop down menu for the pen tool to select the line type is offsetting the drop down menu by a few centimeters away from where it should be.

**To Reproduce**
I'm not 100% sure how to reproduce this behaviour, I just clicked on the drop down menu and saw that it was offset. It doesn't usually do that.

**Expected behavior**
I expect the drop down menu to drop down from directly where the drop down menu icon is located

**Screenshots of Problem**
![signal-2022-02-08-191745](https://user-images.githubusercontent.com/7829231/153115371-2c98c85d-e542-4c50-8035-3787befba7d0.jpeg)

**Comment by K4LCIFER** [comment]
The issue appears to be persistant. I first restarted xournal to no avail. I then rebooted my computer, and the problem still persists. Not sure whats causing this new behaviour.

**Comment by rolandlo** [comment]
Do you use an external monitor and/or fractional scaling? I can't reproduce the issue (on Ubuntu 21.10), neither with the `.deb` package (using Gtk 3.24.30) nor with the flatpack (which uses Gtk 3.24.31 like you do).

**Comment by K4LCIFER** [comment]
> Do you use an external monitor and/or fractional scaling?

I have 3 displays: 1 1440p main monitor, 1 1080p secondary monitor 1 1080p drawing tablet. Neither display is using fractional scaling. 

I should also point out that this is new behaviour. I did not notice this behaviour yesterday when using xournal. I have performed a system update so perhaps a dependency is performing differently?

**Comment by rolandlo** [comment]
Does it also happen when you disconnect from the secondary monitor (and the drawing tablet)?
It's very unlikely that we can do anything about it from Xournal++'s side, but we might find out, what dependency upgrade caused the issue.

**Comment by K4LCIFER** [comment]
> Does it also happen when you disconnect from the secondary monitor (and the drawing tablet)?

I tested two combinations:
1. Secondary monitor disconnected leaving main monitor and drawing tablet connected: Problem went away
2. Drawing tablet disconnected leaving main monitor and secdonary monitor connected: Problem went away

It seems that the problem only exists when all 3 monitors are connected. Furthermore, there is inconsistency with what is affected. Currently the only dropdown menus that appear to be effected are the pen drop down and the eraser drop down. All toolbar dropdowns are fine, and the new page drop down is fine.

**Comment by rolandlo** [comment]
> It seems that the problem only exists when all 3 monitors are connected. Furthermore, there is inconsistency with what is affected. Currently the only dropdown menus that appear to be affected are the pen drop down and the eraser drop down. All toolbar dropdowns are fine, and the new page drop down is fine.

Is that independent of the toolbar you use? Which toolbar do you use anyway?

**Comment by K4LCIFER** [comment]
> Is that independent of the toolbar you use?

What do you mean?

> Which toolbar do you use anyway?

I'm not sure. Whichever is default with Xournal++? This one:
![image](https://user-images.githubusercontent.com/7829231/153135121-d879b1ec-cf3c-4eec-9f7f-6d1005fcd62e.png)

**Comment by K4LCIFER** [comment]
> Is that independent of the toolbar you use?

Do you mean independent of the toolbar layout?

> Which toolbar do you use anyway

If you mean what toolbar layout in Xournal++, I use the Right Handed Note Taking layout.

**Comment by rolandlo** [comment]
Yes, I meant the toolbar layout in Xournal++. Does it have an influence which one you choose?

**Comment by K4LCIFER** [comment]
> Does it have an influence which one you choose?

It seems that regardless of the layout chosen, the pen, eraser, selection, and shape dropdowns all exhibit the behaviour as described in the issue, but the new page drop down consistently works as is expected.

**Comment by alexhuntley** [comment]
I'm experiencing the same issue. It seems to depend on some state in my system, because sometimes the dropdowns work normally, and this persists even after restarting Xournal++. When they are offset, it seems they can be fixed by removing and then reconnecting a display.

OS: Fedora 34
Kernel: 5.15.6-100.fc34.x86_64
DE: GNOME 40.5 on Wayland
Xournal++: 1.1.0 (from the Fedora repositories)
Displays: 1 internal laptop @ 1080p, 100% scaling, 1 external via HDMI @ 1080p, 100% scaling

**Comment by K4LCIFER** [comment]
Just to update this, I have since switched from Gnome to KDE, and I still have the same issue.

- OS: Arch Linux, Kernel v6.4.7-arch1-1
- DE: KDE 5.27.6, Wayland
- Xournal++: v1.2.0
- GTK: v3.24.38
- Displays: 1440p (100% scaling, Display Port), 1080p (100% scaling, HDMI), 1080p (100% scaling, HDMI)
